.github/workflows/ubuntu_focal.yml

```diff
@@ -20,20 +26,20 @@ jobs:
       - name: Install dependencies
         working-directory: workspace
         run: |
-          pwd
-          sudo apt update -q
-          sudo apt install -q -y clang-tidy python3 python3-pip
-          sudo pip3 install -q --upgrade pip
-          sudo pip3 install -q colcon-common-extensions rosdep vcstool
-          sudo rosdep init -q
+          export DEBIAN_FRONTEND=noninteractive
+          apt update -q
+          apt install -q -y clang-tidy python3 python3-pip git
+          pip3 install -q --upgrade pip
+          pip3 install -q colcon-common-extensions rosdep vcstool
+          rosdep init -q
           rosdep update -q
-          vcs import --input "${{ github.workspace }}/workspace/src/boost_plugin_loader/dependencies.repos" src/
+          vcs import src < src/boost_plugin_loader/dependencies.repos
           rosdep install --from-paths src --ignore-src -r -y -q
 
       - name: Build
         working-directory: workspace
         run: |
-          colcon build --event-handlers console_direct+ --cmake-args -DCMAKE_BUILD_TYPE=Debug -DBUILD_TESTING=ON -DENABLE_CLANG_TIDY=ON -DENABLE_CODE_COVERAGE=ON
```

Comment:
When using clang-tidy and code coverage require debug symbols and I have had challenges with release with debug producing consistent results. This was the main reason why they have been separate CI builds.

Reply:
Interesting. I can look into setting the CMake arguments conditionally based on the event type.

Reply:
Fixed this by adding a step that will conditionally change the build type to release if it is a release job